app/models/location.rb

```diff
@@ -131,9 +131,18 @@ def self.find_or_new_by_fields(loc_info)
       success, resp = geosearch_by_osm_id(loc_info[:osm_id], loc_info[:osm_type])
       raise "Couldn't fetch OSM ID #{loc_info[:osm_id]} (#{loc_info[:osm_type]})" unless success
 
-      resp = LocationHelper.adapt_location_iq_response(resp)
-      # 'New' without saving so make sure caller saves.
-      new_from_params(resp)
+      location_fields = LocationHelper.adapt_location_iq_response(resp)
+
+      # Surprisingly, it is possible that adapt_location_iq_response returns location_fields
+      # that are different from loc_info above (where loc_info is passed in from the front-end).
```

Comment:
did you find a reason for this surprise? or, did you find what fields differed unexpectedly?

Reply:
I broke down the API inconsistencies I discovered here: https://czi.quip.com/OZw6ATNbMyJi/LocationIQ-Gotchas